internal/server/statvar/statvar_group.go

```diff
@@ -114,6 +116,52 @@ func filterSVG(in *pb.StatVarGroups, statVars []string) *pb.StatVarGroups {
 	return result
 }
 
+// fixCustomRootDescendentStatVarCount returns the DescendentStatVarCount for node curID.
+// Mutates DescendentStatVarCount for all nodes in id2Node starting at customSvgRoot.
+func fixCustomRootDescendentStatVarCount(id2Node map[string]*pb.StatVarGroupNode, curID string) int32 {
+	curNode, ok := id2Node[curID]
+	if !ok {
+		return 0
+	}
+	var descendentStatVarCount int32
+	for _, childSVG := range curNode.GetChildStatVarGroups() {
+		// Child SVG protos have its own DescendentStatVarCount that must be set.
+		childSVG.DescendentStatVarCount = fixCustomRootDescendentStatVarCount(id2Node, childSVG.GetId())
+		descendentStatVarCount += childSVG.DescendentStatVarCount
+	}
+	curNode.DescendentStatVarCount = descendentStatVarCount + int32(len(curNode.ChildStatVars))
+	return curNode.GetDescendentStatVarCount()
+}
+
+// mergeCustomSVGNode merges n2's svs and child svgs into n1.
+// Merge here refers to set operation.
+// Warning: n1's DescendentStatVarCount is unchanged.
+// Caller is responsible for calling fixCustomRootDescendentStatVarCount
+// at the end of all merges.
+func mergeCustomSVGNode(n1, n2 *pb.StatVarGroupNode) {
```

Comment:
Can you use this function https://github.com/datacommonsorg/mixer/blob/master/internal/util/util.go#L289 to merge child SVG and child SV

Reply:
I tried but I'm not sure if it was cleaner.  The MergeDedupe function merges strings but here we need to creating a string slice list of just the ids, a reference of the id -> child SVG and id -> child SV, and then loop through eveything again. We could try to make MergeDedup to use Generics but that may be for later.

Reply:
I see, here try to merge object using ID to dedup.